hips/hip-9999.md

```diff
@@ -0,0 +1,79 @@
+---
+hip: 9999
+title: "Add Repository URL and tarball digest to a charts release metadata"
+authors: [ "Luke Reed <luke@lreed.net>" ]
+created: "2021-11-24"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Currently there is no way of knowing the exact repository that a chart originated from for a given release. There are ways to guess by looking at the sources list in the chart metadata, but since this is provided via the `Chart.yaml` file, any forks of that repository will look identical. The only way to concretely know what the repository URL is that was used at install/upgrade time is to populate a repository URL metadata field in the release during the install or upgrade. It should work for both traditional repositories behind an http web server, as well as an OCI compliant repository.
+
+The second part of this proposal would include the tarball digest of a chart in the metadata. This aspect is less fleshed out in code yet, but would be a nice alternative for installing a local chart from a tarball when the repository URL would not be known. The reason this is useful is because it can be used when interacting with other third-party databases, such as artifact-hub.
+
```

Comment:
What will be done for an installation done from a local filesystem directory?

Reply:
Currently, in my reference implementation, the field would not get populated and would be omitted from the release metadata.